dev-python/qscintilla-python/qscintilla-python-2.13.3.ebuild

```diff
@@ -21,60 +21,78 @@ S=${WORKDIR}/${MY_P}/Python
 LICENSE="GPL-3"
 SLOT="0"
 KEYWORDS="amd64 ~arm64 ~ppc64 x86"
-IUSE="debug"
+IUSE="debug +qt5 qt6"
 
-REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+REQUIRED_USE="|| ( qt5 qt6 ) ${PYTHON_REQUIRED_USE}"
+
+# no tests
+RESTRICT="test"
 
 DEPEND="${PYTHON_DEPS}
-	>=dev-python/PyQt5-5.15.5[gui,printsupport,widgets,${PYTHON_USEDEP}]
-	dev-qt/qtcore:5
-	dev-qt/qtgui:5
-	dev-qt/qtprintsupport:5
-	dev-qt/qtwidgets:5
-	~x11-libs/qscintilla-${PV}:=
+	qt5? (
+		>=dev-python/PyQt5-5.15.5[gui,printsupport,widgets,${PYTHON_USEDEP}]
+		dev-qt/qtcore:5
+		dev-qt/qtgui:5
+		dev-qt/qtprintsupport:5
+		dev-qt/qtwidgets:5
+	)
+	qt6? (
+		dev-python/PyQt6[gui,printsupport,widgets,${PYTHON_USEDEP}]
+		dev-qt/qtbase:6[cups,gui,widgets]
+	)
+	~x11-libs/qscintilla-${PV}:=[qt5=,qt6=]
 "
 RDEPEND="${DEPEND}
-	>=dev-python/PyQt5-sip-12.9:=[${PYTHON_USEDEP}]
+	qt5? ( >=dev-python/PyQt5-sip-12.9:=[${PYTHON_USEDEP}] )
+	qt6? ( >=dev-python/PyQt6-sip-13.5:=[${PYTHON_USEDEP}] )
 "
 BDEPEND="
 	>=dev-python/PyQt-builder-1.10[${PYTHON_USEDEP}]
 	>=dev-python/sip-6.2[${PYTHON_USEDEP}]
-	dev-qt/qtcore:5
+	qt5? ( dev-qt/qtcore:5 )
+	qt6? ( dev-qt/qtbase:6 )
 "
 
-src_prepare() {
-	default
-	mv pyproject{-qt5,}.toml || die
+pkg_setup() {
+	MULTIBUILD_VARIANTS=( $(usev qt5) $(usev qt6) )
 }
 
 src_configure() {
-	configuration() {
-		local myconf=(
-			sip-build
-			--verbose
-			--build-dir="${BUILD_DIR}"
-			--scripts-dir="$(python_get_scriptdir)"
-			--qmake="$(qt5_get_bindir)"/qmake
-			--no-make
-			$(usev debug '--debug --qml-debug --tracing')
-		)
-		echo "${myconf[@]}"
-		"${myconf[@]}" || die
+	my_src_configure() {
+		configuration() {
+			local myconf=(
+				sip-build
+				--verbose
+				--build-dir="${BUILD_DIR}"
+				--scripts-dir="$(python_get_scriptdir)"
+				--qmake="/usr/bin/${QMAKE}"
+				--no-make
+				$(usev debug '--debug --qml-debug --tracing')
+			)
+			echo "${myconf[@]}"
+			"${myconf[@]}" || die
 
-		# Run eqmake to respect toolchain and build flags
-		run_in_build_dir eqmake5 -recursive ${MY_PN}.pro
+			# Run eqmake to respect toolchain and build flags
+			run_in_build_dir "${QMAKE}" -recursive ${MY_PN}.pro
+		}
+		case ${MULTIBUILD_VARIANT} in
+			qt5) local QMAKE=eqmake5 ;;
+			qt6) local QMAKE=qmake6 ;;
```

Comment:
Same here. Perhaps we should wait till we have a `eqmake6` function.

Reply:
sure, let's wait for https://github.com/gentoo/gentoo/pull/31254 to include `eqmake6` ...